File: SuperShortcutsPhone/src/main/java/net/geekstools/supershortcuts/PRO/SplitShortcuts/SplitServices/SplitTransparentPair.java
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 5/4/20 1:17 PM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */

package net.geekstools.supershortcuts.PRO.SplitShortcuts.SplitServices;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.view.accessibility.AccessibilityEvent;
import android.view.accessibility.AccessibilityManager;
import android.widget.Button;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import net.geekstools.supershortcuts.PRO.R;
import net.geekstools.supershortcuts.PRO.SecurityServices.Protection;
import net.geekstools.supershortcuts.PRO.SecurityServices.SecurityServicesProcess;
import net.geekstools.supershortcuts.PRO.Utils.Functions.FunctionsClass;

public class SplitTransparentPair extends AppCompatActivity {

    FunctionsClass functionsClass;

    static String splitPackageOne, splitPackageTwo = "";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        SecurityServicesProcess securityServicesProcess = new SecurityServicesProcess(SplitTransparentPair.this);

        if (securityServicesProcess.securityServiceEnabled()) {

            securityServicesProcess.protectIt(getIntent().getStringExtra(Intent.EXTRA_TEXT), new Protection() {

                @Override
                public void processNotProtected() {

                    splitProcess();

                }

                @Override
                public void processProtected() {

                    Toast.makeText(getApplicationContext(), getString(R.string.notAuthorized), Toast.LENGTH_LONG).show();

                    SplitTransparentPair.this.finish();

                }

            });

        } else {

            splitProcess();

        }

    }

    void splitProcess() {

        try {

            functionsClass = new FunctionsClass(getApplicationContext());

            Window window = getWindow();
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
            window.getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR);
            window.setStatusBarColor(Color.TRANSPARENT);
            getWindow().setNavigationBarColor(Color.TRANSPARENT);

            if (!functionsClass.AccessibilityServiceEnabled() && !functionsClass.SettingServiceRunning(SplitScreenService.class)) {

                functionsClass.AccessibilityService(this, true);

            } else {

                final AccessibilityManager accessibilityManager = (AccessibilityManager) getSystemService(ACCESSIBILITY_SERVICE);

                if (getIntent().getAction().equals("load_split_action_pair")) {

                    splitPackageOne = getIntent().getStringArrayExtra("packages")[0];
                    splitPackageTwo = getIntent().getStringArrayExtra("packages")[1];

                } else if (getIntent().getAction().equals("load_split_action_pair_shortcut")) {

                    String categoryName = getIntent().getStringExtra(Intent.EXTRA_TEXT);

                    splitPackageOne = functionsClass.readFileLine(categoryName)[0];
                    splitPackageTwo = functionsClass.readFileLine(categoryName)[1];

                }

                AccessibilityEvent accessibilityEvent = AccessibilityEvent.obtain();
                accessibilityEvent.setSource(new Button(getApplicationContext()));
                accessibilityEvent.setEventType(AccessibilityEvent.TYPE_WINDOW_STATE_CHANGED);
                accessibilityEvent.setAction(10296);
                accessibilityEvent.setClassName(SplitTransparentPair.class.getSimpleName());
                accessibilityEvent.getText().add(getPackageName());
                accessibilityManager.sendAccessibilityEvent(accessibilityEvent);

                Intent splitOne = getPackageManager().getLaunchIntentForPackage(splitPackageOne);
                splitOne.addFlags(
                        Intent.FLAG_ACTIVITY_LAUNCH_ADJACENT |
                                Intent.FLAG_ACTIVITY_NEW_TASK);

                startActivity(splitOne);

                functionsClass.Toast(functionsClass.appName(splitPackageOne), Gravity.TOP);

            }

        } catch (Exception e) {
            e.printStackTrace();
        }

    }

    @Override
    public void onPause() {
        super.onPause();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();

        SplitTransparentPair.this.finish();

    }

}

File: SuperShortcutsPhone/src/main/java/net/geekstools/supershortcuts/PRO/SplitShortcuts/SplitServices/SplitTransparentPair.kt
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 5/4/20 1:17 PM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */
package net.geekstools.supershortcuts.PRO.SplitShortcuts.SplitServices

import android.content.Intent
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.WindowManager
import android.view.accessibility.AccessibilityEvent
import android.view.accessibility.AccessibilityManager
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import net.geekstools.supershortcuts.PRO.R
import net.geekstools.supershortcuts.PRO.SecurityServices.Protection
import net.geekstools.supershortcuts.PRO.SecurityServices.SecurityServicesProcess
import net.geekstools.supershortcuts.PRO.Utils.Functions.FunctionsClass
import net.geekstools.supershortcuts.PRO.Utils.Functions.SystemInformation

class SplitTransparentPair : AppCompatActivity() {

    companion object {
        var splitPackageOne = ""
        var splitPackageTwo = ""
    }

    val systemInformation by lazy {
        SystemInformation(applicationContext)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val securityServicesProcess = SecurityServicesProcess(this@SplitTransparentPair)

        if (securityServicesProcess.securityServiceEnabled()) {

            securityServicesProcess.protectIt(intent.getStringExtra(Intent.EXTRA_TEXT)!!,
                object : Protection {

                    override fun processNotProtected() {
                        splitProcess()
                    }

                    override fun processProtected() {
                        Toast.makeText(applicationContext, getString(R.string.notAuthorized), Toast.LENGTH_LONG).show()

                        this@SplitTransparentPair.finish()

                    }

                })

        } else {

            splitProcess()

        }
    }

    fun splitProcess() {
        try {

            val functionsClass: FunctionsClass = FunctionsClass(applicationContext)

            val window = window
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
            window.clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS)
            window.decorView.systemUiVisibility = View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR
            window.statusBarColor = Color.TRANSPARENT
            getWindow().navigationBarColor = Color.TRANSPARENT

            if (!functionsClass.AccessibilityServiceEnabled() && !functionsClass!!.SettingServiceRunning(SplitScreenService::class.java)) {

                functionsClass.AccessibilityService(this, true)

            } else {
                val accessibilityManager = getSystemService(ACCESSIBILITY_SERVICE) as AccessibilityManager

                if (intent.action == "load_split_action_pair") {

                    splitPackageOne = intent.getStringArrayExtra("packages")!![0]
                    splitPackageTwo = intent.getStringArrayExtra("packages")!![1]

                } else if (intent.action == "load_split_action_pair_shortcut") {

                    val categoryName = intent.getStringExtra(Intent.EXTRA_TEXT)

                    splitPackageOne = functionsClass.readFileLine(categoryName)!![0]
                    splitPackageTwo = functionsClass.readFileLine(categoryName)!![1]

                }

                val accessibilityEvent = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
                    AccessibilityEvent()
                } else {
                    AccessibilityEvent.obtain()
                }
                accessibilityEvent.setSource(Button(applicationContext))
                accessibilityEvent.eventType = AccessibilityEvent.TYPE_WINDOW_STATE_CHANGED
                accessibilityEvent.action = 10296
                accessibilityEvent.className = SplitTransparentPair::class.java.simpleName
                accessibilityEvent.text.add(packageName)

                accessibilityManager.sendAccessibilityEvent(accessibilityEvent)

                val splitIntent = packageManager.getLaunchIntentForPackage(SplitTransparentPair.splitPackageTwo)
                splitIntent?.addFlags(
                    Intent.FLAG_ACTIVITY_LAUNCH_ADJACENT or
                            Intent.FLAG_ACTIVITY_NEW_TASK
                )
                startActivity(splitIntent)
                Log.d(this@SplitTransparentPair.javaClass.simpleName, "Split It: ${SplitTransparentPair.splitPackageTwo}")

            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    public override fun onPause() {
        super.onPause()
    }

    public override fun onDestroy() {
        super.onDestroy()

        this@SplitTransparentPair.finish()

    }

}
